include/mbedtls/ssl.h

```diff
@@ -2721,16 +2726,31 @@ int mbedtls_ssl_conf_own_cert( mbedtls_ssl_config *conf,
  *                 of the SSL configuration.
  *
  * \return         \c 0 if successful.
- * \return         An \c MBEDTLS_ERR_SSL_XXX error code on failure.
+ * \return         #MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE if no more PSKs
+ *                 can be configured. In this case, the SSL configuration
+ *                 remains usable, but the PSK has not been configured.
+ * \return         Another negative error code on other kinds of failure.
  */
 int mbedtls_ssl_conf_psk( mbedtls_ssl_config *conf,
                 const unsigned char *psk, size_t psk_len,
                 const unsigned char *psk_identity, size_t psk_identity_len );
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
 /**
- * \brief          Configure an opaque pre-shared key (PSK) and identity
- *                 to be used in PSK-based ciphersuites.
+ * \brief          Configure one or more opaque pre-shared keys (PSKs) and
+ *                 their identities to be used in PSK-based ciphersuites.
+ *
+ *                 This function may be called multiple times to attempt
```

Comment:
The following description would have my preference I think:
```
This function may be called multiple times to attempt to register multiple PSKs.
Once the limit is reached, this function fails, maintaining the PSKs previously
configured and ignoring the excess request.
```
Thus leaving the maximum number of supported PSK to a separate note (by the way, defining a macro for the maximum number could help for this documentation) and removing the project history part that to me does not really belong here.

Reply:
@ronald-cron-arm This is my preference, too, but not that of @gilles-peskine-arm and @mpg. Are you OK if we go with proposal (e) from above?

Reply:
I was just trying to help here. Please go ahead.